Add code of conduct

Tieqiong opened the pull request:
Do we need this? I mentioned it in readme

sbillinge commented:
@Tieqiong this looks good.  Please can you confirm where you got this from and any changes you made to it (if any) from the original version.  Thanks.   This deals with somewhat delicate issues so we want to get it correct.

Tieqiong commented:
@sbillinge I got this from [CONTRIBUTOR COVENANT CODE OF CONDUCT](https://www.contributor-covenant.org/version/2/1/code_of_conduct/). I modified it from the .md format to fit the .rst syntax (namely text levels and hyperlinks) but the content should remain effectively the same. I also added your email (sb2896@columbia.edu) in the Enforcement section as the community leaders 
>...
>Enforcement
>
>Instances of abusive, harassing, or otherwise unacceptable behavior may be
>reported to the community leaders responsible for enforcement at
>sb2896@columbia.edu. All complaints will be reviewed and investigated promptly and fairly. ...
>

sbillinge commented:
OK, sounds good.  Is the content "effectively the same" or "the same".  If the later we can go with that.

Tieqiong commented:
The content and hierarchy are the same, only format difference (line space etc.) and hyperlink style (blue underline etc.) are different
